File: demos/php/dsa/linear/dynamic_array.php
<?php
echo "=== PHP Dynamic Arrays & SplDoublyLinkedList ===\n";
$arr = [10, 20];
$arr[] = 30;
assert(count($arr) === 3);
assert($arr[1] === 20);

$dll = new SplDoublyLinkedList();
$dll->push("tail");
$dll->unshift("head");
assert($dll->bottom() === "head");
assert($dll->top() === "tail");

echo "PHP Array count: " . count($arr) . ", DLL count: " . $dll->count() . "\n";
echo "PHP Dynamic Array tests passed successfully.\n";

File: demos/rust/dsa/linear/dynamic_array.rs
fn main() {
    println!("=== Rust Vec & VecDeque ===");
    let mut vec: Vec<i32> = Vec::with_capacity(2);
    vec.push(10);
    vec.push(20);
    vec.push(30);

    assert_eq!(vec.len(), 3);
    assert!(vec.capacity() >= 4);
    assert_eq!(vec[0], 10);
    assert_eq!(vec.pop(), Some(30));

    let sum: i32 = vec.iter().sum();
    assert_eq!(sum, 30);
    println!("Rust Vec len={}, sum={}", vec.len(), sum);
    println!("Rust Dynamic Array tests passed successfully.");
}
